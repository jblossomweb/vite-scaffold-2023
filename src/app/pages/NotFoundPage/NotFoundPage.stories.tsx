import React from 'react'
import { MemoryRouter } from 'react-router'
import type { Meta, StoryObj } from '@storybook/react'

import NotFoundPage from './NotFoundPage'

const meta = {
  title: 'pages/NotFoundPage',
  component: NotFoundPage,
  parameters: {
    layout: 'fullscreen',
  },
  render: () => (
    <MemoryRouter>
      <NotFoundPage />
    </MemoryRouter>
  ),
} satisfies Meta<typeof NotFoundPage>

export default meta
type Story = StoryObj<typeof meta>

export const Sample: Story = {
  args: {},
}
