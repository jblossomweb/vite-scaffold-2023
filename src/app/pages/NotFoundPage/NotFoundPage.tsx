import React from 'react'
import { Link } from 'react-router-dom'

import HeroTemplate from 'app/templates/HeroTemplate'

const NotFoundPage: React.FC = () => (
  <HeroTemplate title="404: Not Found">
    <h1>Sorry! That page was not found.</h1>
    <p>
      <Link to="/">Back to Home</Link>
    </p>
  </HeroTemplate>
)

export default NotFoundPage
