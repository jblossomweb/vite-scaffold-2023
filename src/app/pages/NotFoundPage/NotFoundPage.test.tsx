import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render, screen } from '@testing-library/react'
import storySnaps from 'test/utils/storySnaps'

import NotFoundPage from './NotFoundPage'
import * as stories from './NotFoundPage.stories'

describe('app/pages/NotFoundPage', () => {
  describe('stories', () => {
    storySnaps(stories)
  })

  describe('assertions', () => {
    const renderScreen = () =>
      render(
        <MemoryRouter>
          <NotFoundPage />
        </MemoryRouter>,
      )

    it('renders back to home link', () => {
      renderScreen()
      const backHome = screen.getByText(/back to home/i)
      expect(backHome).toBeInTheDocument()
    })
  })
})
