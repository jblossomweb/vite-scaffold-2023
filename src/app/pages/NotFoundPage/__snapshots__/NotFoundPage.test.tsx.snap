// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/pages/NotFoundPage stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center"
  >
    <header
      class="min-h-screen flex flex-col items-center justify-center"
    >
      <h1>
        Sorry! That page was not found.
      </h1>
      <p>
        <a
          href="/"
        >
          Back to Home
        </a>
      </p>
    </header>
  </div>
</DocumentFragment>
`;
